brave/brave-core
# Add rewards notification to backup wallet

Fixes brave/brave-browser#1164
Requires brave/brave-ui#312

Notify the user that they need to backup their wallet if they haven't done so for 7 days after funding it. Backup can be completed by going to the wallet properties in Rewards settings and selecting either Copy, Print, or Save to File. If notification is ignored, another notification is shown every 7 days.

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:

Each test should start with a clean profile.

- You can set `brave.rewards.backup_notification_frequency` (units are in microseconds) in your profile's `Preferences` file to shorten the backup notification interval (defaults to 7 days).
- You can set `brave.rewards.notification_timer_interval` (units are in microseconds) in your profile's `Preferences` file to shorten the interval that we wait before checking if notifications need to be shown (defaults to 15 minutes).

### Test 1

- Delete any existing profile
- Launch Brave
- Enable Rewards
- Don't add any funds to wallet
- Wait 7 days (if you don't have 7 days to spare, see tip above for shortening this interval)
- Now wait > 15 minutes for notification check (or see tips above to shorten this interval)
- Rewards icon should _not_ be badged
- `brave.rewards.backup_succeeded` should be set to `false` in `Preferences` for profile

### Test 2

- Delete any existing profile
- Launch Brave
- Enable Rewards
- Add funds to wallet
- Wait 7 days (if you don't have 7 days to spare, see tip above for shortening this interval)
- Now wait > 15 minutes for notification check (or see tips above to shorten this interval)
- Rewards icon should be badged and should contain a notification like this:

![image](https://user-images.githubusercontent.com/5033691/49668665-d7279280-fa2c-11e8-9fb2-2a09eb701755.png)

- Clear notification
- Bring up wallet properties dialog and select Copy or Print or Save to File
- Wait 7 days again (if you don't have 7 days to spare, see tip above for shortening this interval)
- Wait > 15 minutes again for next notification check (or see tips above to shorten this interval)
- Rewards icon should _not_ rebadge
- `brave.rewards.backup_succeeded` should be set to `true` in `Preferences` file for profile

### Test 3

- Delete any existing profile
- Launch Brave
- Enable Rewards
- Add funds to wallet
- Wait 7 days (if you don't have 7 days to spare, see tip above for shortening this interval)
- Now wait > 15 minutes for notification check (or see tips above to shorten this interval)
- Rewards icon should be badged and should contain a notification like this:

![image](https://user-images.githubusercontent.com/5033691/49668665-d7279280-fa2c-11e8-9fb2-2a09eb701755.png)

- Clear notification
- Don't backup wallet
- Wait 7 days again (if you don't have 7 days to spare, see tip above for shortening this interval)
- Wait > 15 minutes again for next notification check (or see tips above to shorten this interval)
- Rewards icon _should_ rebadge
- `brave.rewards.backup_succeeded` should be set to `false` in `Preferences` file for profile

## Reviewer Checklist:
